.jenkins/nightly.groovy

```diff
@@ -0,0 +1,41 @@
+pipeline {
+    agent none
+
+    stages {
+        stage('HIP-ROCm-3.10-C++14') {
+            agent {
+                dockerfile {
+                    filename 'Dockerfile.hip'
+                    dir 'scripts/docker'
+                    additionalBuildArgs '--build-arg BASE=rocm/dev-ubuntu-20.04:3.10'
+                    label 'rocm-docker && vega'
+                    args '-v /tmp/ccache.kokkos:/tmp/ccache --device=/dev/kfd --device=/dev/dri --security-opt seccomp=unconfined --group-add video --env HIP_VISIBLE_DEVICES=$HIP_VISIBLE_DEVICES'
+                }
+            }
+            steps {
+                sh '''rm -rf kokkos &&
+                      git clone -b develop https://github.com/kokkos/kokkos.git && cd kokkos && \
+                      mkdir build && cd build && \
+                      cmake \
+                        -DCMAKE_CXX_COMPILER=hipcc \
+                        -DKokkos_ENABLE_HIP=ON \
+                        -DKokkos_ARCH_VEGA906=ON \
+                        .. && \
+                       make -j8 && make install && \
+                       cd ../.. && rm -rf kokkos'''
+                sh '''rm -rf build && mkdir -p build && cd build && \
+                      cmake \
+                        -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+                        -DCMAKE_CXX_COMPILER=hipcc \
```

Comment:
Shouldn't there be a `-DKokkos_DIR=<path-to-kokkos-install>` to point to the Kokkos install?

Reply:
I actually really dislike the using `<PackageName>_DIR` in to stir the search procedure in CMake.  I could have `-DKokkos_ROOT=/opt/kokkos` but I chose to install it there in part because I knew it was going to fall into the set of possible prefixes that CMake would construct for Unix.

Reply:
After chatting with Nathan I realize `-DKokkos_ROOT` wouldn't work because KokkosKernels currently requires a minimum cmake version of 3.10 but this would work when we bump to 3.12+